src/app/shared/services/song.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable }    from '@angular/core';
+
+import { Observable } from "rxjs/Observable";
+
+import { ApiService } from './api.service';
+import { User } from '../models/song'
+
+import 'rxjs/add/operator/map';
+
+@Injectable()
+export class SongService {
+  private songsUrl = '/songs';
+
+  constructor(private apiService: ApiService) {
+  }
+
+  getSongs(): Observable<Song[]> {
+    return this.apiService.get(`${this.songsUrl}`)
+      .map((res: any) => res.data as Song[]);
```

Comment:
Based on the changes in master, this will need to change to the following:
```
      .map((res: Response) => {
        const resBody: any = res.json();
        return resBody.songs as Song[];
      });
```